Landing Page

jheel05 opened the pull request:
This PR contains the Static Changes of the landing page, i.e, NavBar, CategoriesBar, Image carousel, Footer.

Commit:
static chngs

Commit:
drop btn chng

Commit:
changes made and PR comments addresed.

jheel05 committed:
Delete src/components/PoductsList directory

jheel05 commented:
The PR comments have been addressed, and changed the file structure.

srivsaks commented:
Please add screenshots/video recoridng of the changes

srivsaks commented on src/App.jsx:
```diff
@@ -1,13 +1,11 @@
-
-import './App.css'
+import LandingPage from './pages/landingpage.jsx'
```
Install prettier and run that on all your files.

jheel05 replied:
Resolved

srivsaks commented on src/components/Login/login.jsx:
```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
```
Remove this file from the PR if its not ready or else finish this one and push the new changes in the next round of revision.

jheel05 replied:
Resolved

srivsaks commented on src/components/ImageSlider/imageslider.jsx:
```diff
@@ -0,0 +1,70 @@
+  const [currentImage, setCurrentImage] = useState(0); 
+  const [isHovering, setIsHovering] = useState(false);
+  const nextImage = () => {
+    setCurrentImage((prevImage) => (prevImage + 1) % images.length);
```
images.length could have been a variable and then used at places.

jheel05 replied:
Resolved

srivsaks commented on src/components/ImageSlider/imageslider.jsx:
```diff
@@ -0,0 +1,70 @@
+    return()=>{
+      clearInterval(intervalTimer);
+    }
+  })
```
why no dependency? do we want to make this useEffect run on every render?

jheel05 replied:
Resolved

srivsaks commented on src/components/ImageSlider/imageslider.jsx:
```diff
@@ -0,0 +1,70 @@
+
+      {/* Slider button container */}
+      <div className='w-full h-fit flex justify-between absolute left-0 top-[50%] translate-y-[-50%]'>
+        <span onClick={prevImage} className='bg-white shadow-[0_3px_10px_rgb(0,0,0,0.2)] px-[2px] py-3 sm:px-1 sm:py-5 lg:px-3 lg:py-9 rounded-r-sm cursor-pointer'><FaAngleLeft /></span>
+        <span onClick={nextImage} className='bg-white shadow-[0_3px_10px_rgb(0,0,0,0.2)] px-[2px] py-3 sm:px-1 sm:py-5 lg:px-3 lg:py-9 rounded-l-sm cursor-pointer'><FaAngleRight /></span>
+      </div>
```
Convert it to a reusable component and then use it.

srivsaks replied:
Also why not button element? why a span element ?

jheel05 replied:
Resolved

srivsaks commented on src/components/Footer/footer.jsx:
```diff
@@ -0,0 +1,183 @@
+      <div id="footer">
+        <div>
+          <li className="heading">ABOUT</li>
+          <li>
+            <a
+              href="https://corporate.flipkart.net/corporate-home"
+              target="_blank"
+              >About Us</a
+            >
+          </li>
```
Make this a re-usable component.Applicable at other places too.

jheel05 replied:
Resolved

srivsaks commented on src/utils/search.js:
```diff
@@ -0,0 +1,38 @@
+let timer;
```
convert it into react code and not js.

jheel05 replied:
Resolved.

Commit:
Updated landing page

jheel05 commented:
Updated the Landing Page. Addressed all previous comments on the PR. Now includes header, footer, product list display by categories(using API), image carousel, search function(using API), and login functionality(using API). Check it out!

Screenshots:
<img width="1512" alt="Screenshot 2024-03-06 at 2 07 54 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/75b772a1-368d-47f9-aa2c-58cefdeee66b">
<img width="1512" alt="Screenshot 2024-03-06 at 2 07 57 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/042f3f9c-e50b-4498-a340-a0a6fdc27079">
<img width="1512" alt="Screenshot 2024-03-06 at 2 08 00 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/112e226e-6263-4e6c-b062-a2e81a346233">
<img width="1512" alt="Screenshot 2024-03-06 at 2 08 08 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/2d8058c0-eb8f-4fda-9853-5df6c17d5f1f">
<img width="1512" alt="Screenshot 2024-03-06 at 2 08 15 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/2b626d6f-b0c5-4ab2-90b4-107b5807c4e2">
<img width="1512" alt="Screenshot 2024-03-06 at 2 08 24 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/1203c302-3c1e-4134-babf-c2b2f7df5883">
<img width="1512" alt="Screenshot 2024-03-06 at 2 08 35 PM" src="https://github.com/jheel05/Flipkart-Clone-React-Phase-3/assets/73063350/cf8aade7-6340-40e2-869f-f40b56d81a52">

srivsaks commented on src/API/CategoryProducts/CategoryProductAPI.js:
```diff
@@ -0,0 +1,12 @@
+// api.js
```
nit: you can remove this.

jheel05 replied:
Resolved

srivsaks commented on src/API/CategoryProducts/CategoryProductAPI.js:
```diff
@@ -0,0 +1,12 @@
+// api.js
```
give folder structure name src/api ....

jheel05 replied:
Resolved
